cmd/deploy.go

```diff
@@ -221,7 +223,29 @@ func autofillApimodel(dc *deployCmd) {
 			// TODO: consider caching the creds here so they persist between subsequent runs of 'deploy'
 			appName := dc.containerService.Properties.MasterProfile.DNSPrefix
 			appURL := fmt.Sprintf("https://%s/", appName)
-			applicationID, servicePrincipalObjectID, secret, err := dc.client.CreateApp(appName, appURL)
+			var replyURLs *[]string
+			var requiredResourceAccess *[]graphrbac.RequiredResourceAccess
+			if dc.containerService.Properties.OrchestratorProfile.OrchestratorType == api.OpenShift {
+				appName = fmt.Sprintf("%s.%s.cloudapp.azure.com", appName, dc.containerService.Properties.AzProfile.Location)
+				appURL = fmt.Sprintf("https://%s:8443/", appName)
+				replyURLs = to.StringSlicePtr([]string{fmt.Sprintf("https://%s:8443/oauth2callback/Azure%%20AD", appName)})
+				ra := []graphrbac.ResourceAccess{
+					{
+						// TODO: where is this UUID defined?
+						ID:   to.StringPtr("311a71cc-e848-46a1-bdf8-97ff7156d8e6"),
+						Type: to.StringPtr("Scope"),
+					},
+				}
+				rra := []graphrbac.RequiredResourceAccess{
+					{
+						// TODO: where is this UUID defined?
+						ResourceAppID:  to.StringPtr("00000002-0000-0000-c000-000000000000"),
```

Comment:
ditto on the constant here.  From what I could find this is always the reserved id for AD in the tenant.  It can be found by listing the service principals:

```
$ az ad sp list -o table

AppId                                 DisplayName                                                   
------------------------------------  ------------------------------------------------------------
00000002-0000-0000-c000-000000000000  Windows Azure Active DirectoryServicePrincipal
```

(output abbreviated)